storage/reads/array_cursor.gen.go

```diff
@@ -298,11 +324,79 @@ func newFloatWindowLimitArrayCursor(cur cursors.FloatArrayCursor, every int64) *
 	}
 }
 
-func (c *floatWindowLimitArrayCursor) Stats() cursors.CursorStats {
+func (c *floatWindowLastArrayCursor) Stats() cursors.CursorStats {
 	return c.FloatArrayCursor.Stats()
 }
 
-func (c *floatWindowLimitArrayCursor) Next() *cursors.FloatArray {
+func (c *floatWindowLastArrayCursor) Next() *cursors.FloatArray {
+	every := c.every
+	if every == 0 {
+		every = math.MaxInt64
+	}
+
+	cur := -1
+
+NEXT:
+	var a *cursors.FloatArray
+
+	if c.tmp.Len() > 0 {
+		a = c.tmp
+	} else {
+		a = c.FloatArrayCursor.Next()
+	}
+
+	if a.Len() == 0 {
+		c.res.Timestamps = c.res.Timestamps[:cur+1]
+		c.res.Values = c.res.Values[:cur+1]
+		return c.res
+	}
+
+	for i, t := range a.Timestamps {
+		if t >= c.final {
+			cur++
+		}
+
+		if cur == MaxPointsPerBlock {
+			c.tmp.Timestamps = a.Timestamps[i+1:]
```

Comment:
Imagine the following case where `every = 2m` and `MaxPointsPerBlock = 1`:
```
0, "value0"
1m, "value1"
2m, "value2"
```
The first window was `[0, 2m)`, we got the first output `1m, "value1"`, `c.final = 2m`, `cur = 0`
Then we encountered `2m, "value2"`. Because `t >= c.final`, we should increment `cur`  and that reached `MaxPointsPerBlock = 1`, then we assigned `tmp` to `data[i+1:]`. In this case `data[i]` is already the last record, so `data[i+1:]` is actually an empty slice.

The next time when we call `Next()`, `a.tmp.Len() == 0`, `c.FloatArrayCursor.Next()` also returns a 0-length array, so in line 349, you return `c.res.Timestamps[:cur+1]` and `c.res.Values[:cur+1]` which is basically `c.res.Timestamps[:0]` and `c.res.Values[:0]`, they are both empty.

Now, where is my `2m, "value2"`?

Reply:
Typo.
```suggestion
			c.tmp.Timestamps = a.Timestamps[i:]
```

Reply:
Some test coverage for it will be desirable

Reply:
Absolutely. Changes incoming...